### ISSUE TITLE
Is this module published to NPM?

### ISSUE THREAD

**Issue by mykola-mokhnach** [nut-tree/opencv4nodejs#13]
Hello,

First of all we, the Appium developers team, would like to thank you for supporting and maintaining the fork of opencv4nodejs.
It looks like the original repository is pretty much abandoned and we'd like to know if you have any plans to publish your fork on NPM, so we could use it in Appium instead of the original (unmaintained) one. Thanks.

**Comment by s1hofmann** [comment]
Hi @mykola-mokhnach 👋 

Yes, I have it published on npm, but before you start using it, may I ask you for a little discussion about your planned usage and expectations?
I know about the widespread use of Appium and I started this fork only because I'm providing image matching functionality in my own UI automation / testing framework https://github.com/nut-tree/nut.js, so this would be kind of a big deal since it's not my main focus.

**Comment by mykola-mokhnach** [comment]
Thanks for your response @s1hofmann 

Yes, I know that the purpose of this fork is a bit different from the original lib. Basically, our usage scenario is pretty limited to image comparison/matching. [This](https://github.com/appium/appium-support/blob/master/lib/image-util.js) file basically contains all the functionality we import from opencv4nodejs lib.

Also, Appium does not have a direct dependency to opencv4nodejs, but rather expects it to be installed separately. That is why I was thinking that a possibility of having a prebuilt image would even be better for our case. I'd just have to adjust the client code there to properly load the bundle from the place where it was installed.

**Comment by s1hofmann** [comment]
> Appium does not have a direct dependency to opencv4nodejs, but rather expects it to be installed separately

Assuming only a fraction of the weekly downloads of Appium also install opencv4nodejs would cause quite an increase in traffic.

The problem with prebuilds is the fact that  opencv4nodejs relies on nan bindings. These bindings require new builds for every new node or Electron version. Since opencv4nodejs is huge and I’m only using a small fraction of it I’m actually working on getting rid of it. 
You can read more about it on the [nut.js blog](https://nutjs.dev/blog).
For your usage there’s also the drawback that the precompiled OpenCV does not ship all feature descriptors.

It’s sad to see opencv4nodejs being abandoned, it’s a really powerful library.
I’d love to have it ported to Node-API and I think quite a few of my learnings could be really helpful here, but it’s too much effort for me besides my family, job and my other project.

I just wanted to let you know upfront to avoid any trouble.
Feel free to get in touch with me if you want to discuss this further :)

**Comment by jlipps** [comment]
@s1hofmann thanks for the reply! Since we use OpenCV for basically just template matching as well, it would be interesting to hear what you've decided to move to. I checked out the blog you mentioned and got linked to a TemplateMatcher project which 404s on github--is it open source? We have been looking for a small pure JS library that could do some of the minimal image analysis we need so we can get rid of the opencv bindings altogether (even though it might mean rewriting our internal functions which call those specific APIs).

**Comment by s1hofmann** [comment]
Hi @jlipps 👋 

It's not yet open source as I'm still figuring out a few things and it's still in a draft stage.

It basically boils down to having a single package dedicated for template matching where the actual matching code is written in C++ and exposed as a Node-API add-on.

This setup makes it a lot easier for me to maintain, it ships a recent, precompiled version of OpenCV and bindings are usable across multiple node and Electron versions.

Would this be of interest for you?

**Comment by jlipps** [comment]
Yes, definitely. Pure JS would be nice but a modern Node-API add-on sounds like a huge step forward from where we are.
